src/cascadia/QueryExtension/ILLMProvider.idl

```diff
@@ -0,0 +1,25 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+namespace Microsoft.Terminal.Query.Extension
+{
+    interface ILLMProvider
```

Comment:
Naming nit: should this be `LMProvider` or something along those lines? That would leave flexibility for writing extensions down the road that might not rely on a 'large' model.